### DISCUSSION TITLE
Share learnings from native maps development

### DISCUSSION THREAD

**Discussion by wipfli** [maplibre/maplibre-native#553] [General]
Jesse and Pablo from Microsoft would like to share some learning from their experience with developing native maps SDKs with the MapLibre community. Maarten from TomTom also offered to share their insights.

The question is, what would be interesting topics to learn more about and share experience on?

**Comment by maxammann** [comment]
Renderer related:

* How do you test the renderer on all important platforms? Android, Firefox, Chrome, Linux, Windows, macOS, iOS?
* Do you test it on real hardware or do you usually stick to Android emulators?
* How do you handle different inputs on different platforms? Are you processing raw data or are you using the platform SDKs to get "high level inputs" (like swipes, pinch-to-zoom etc.)


MapLibre related:

* Do you have any current specific pain points with the MapLibre or MapBox styling language?

**Comment by wipfli** [comment]
What are the challenges with text rendering and symbol placement?

**Comment by wipfli** [comment]
How do you measure the performance of a native map renderer?

**Comment by Yatekii** [comment]
How do you render outlines (and lines) efficiently? When I just tesselate once and scale along normals I get heavy artifacts when line thickness get close to line length. Plus, it can be a bit annoying to require normals.

**Reply by maxammann** [reply]
To make the challenge more clear: think about polygons with holes and overlapping polygons.
